### PR TITLE
Dev/mibi/surface closest parameter

### LINKED ISSUE #120: Surface closest parameter
https://github.com/pboyer/verb/blob/3692d7c5f41aab43eef457e74fc0e0513b252cbd/src/verb/geom/NurbsSurface.hx#L252

To implement this we need split surface #123

### PR DESCRIPTION
**Pull request by sonomirco**
### What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [X] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

### Description

This PR introduces the closest parameters, used to define the closest point.
To find the starting value uses split surface instead of the tessellation.

### Related Tickets & Documents

This PR closes #120 #121

### Added tests?

- [X] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [X] 📓 docs
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/GShark.Test.XUnit/VerbTests.cs [hunk 1]
```diff
@@ -365,5 +365,95 @@ public void CurveKnotRefine2()
 
             _testOutput.WriteLine($"{after.controlPoints}");
         }
+
+        [Fact]
```
Comment (sonomirco) [inline src/GShark.Test.XUnit/VerbTests.cs:369]: You can ignore this file, I used to check that everything was working.

File: src/GShark/Operation/Analyze.cs [hunk 2]
```diff
@@ -214,14 +219,193 @@ public static double CurveClosestParameter(ICurve curve, Point3 point)
...
+        /// <param name="point">Point to search from.</param>
+        /// <returns>The closest parameter on the surface.</returns>
+        public static (double u, double v) SurfaceClosestParameter(NurbsSurface surface, Point3 point)
+        {
+            double minimumDistance = double.PositiveInfinity;
+            (double u, double v) selectedUV = (0D, 0D);
+            NurbsSurface splitSrf = surface;
+            double param = 0.5;
+
+            for (int i = 0; i < 2; i++)
```
Comment (sonomirco) [inline src/GShark/Operation/Analyze.cs:236]: I am thinking to iterate 5 times if the results after the first pass are bigger, I gonna break the loop.
@d3ssy @cesarecaoduro

Comment (sonomirco) [reply src/GShark/Operation/Analyze.cs]: Updated

**Review by cesarecaoduro — APPROVED** [review]
All good!

**Pull request merged**